### PR TITLE
add unlimited scrolling list && article pages && script init-db && removed most of the css

### PR DESCRIPTION
**Pull request by lyra-planet**
增加主页列表无限滚动,文章页面

### CODE REVIEW

File: packages/website/.env [hunk 1]
```diff
@@ -1 +1 @@
-../../../../secret/iolab-env
\ No newline at end of file
+DATABASE_URL="file:./dev.db"
```
Comment (xhebox) [inline packages/website/.env:1]: It should be a file, not a redirection link. Delete it and recreate the file with the same content.

File: packages/website/package.json [hunk 2]
```diff
@@ -3,6 +3,9 @@
 	"version": "0.1.0",
 	"sideEffects": false,
 	"scripts": {
+		"init-db": "prisma init --datasource-provider sqlite",
```
Comment (xhebox) [inline packages/website/package.json:6]: That won't work. There is already a directory called `prisma`. You need `prisma generate` instead.

BTW, please provide `"init-db": "run-s init-db:*",` too to run three scripts at once.

File: packages/website/app/components/headerNav.tsx [hunk 3]
```diff
@@ -1,33 +1,43 @@
...
+			w-w='100vw'
+			w-h='50px'
+			w-flex='row'
+			w-top='0'
+			w-items='center'>
+			<div className="left flex">
+				{matches
+					.filter((match) => match.handle && match.handle.breadcrumb)
+					.map((match, index) => (
+						<div className="flex" key={uuid()}>
```
Comment (xhebox) [inline packages/website/app/components/headerNav.tsx:20]:
```suggestion
						<div className="flex" key={match.id}>

```

That is enough, no need for `react-uuid`.

File: packages/website/app/components/headerNav.tsx [hunk 4]
```diff
@@ -1,33 +1,43 @@
...
+	return (
+		<div className="headernav top-0 sticky justify-between flex bg-white items-center py-3 px-2"
+			w-p="12px 10px"
+			w-bg='white'  
+			w-w='100vw'
+			w-h='50px'
+			w-flex='row'
+			w-top='0'
+			w-items='center'>
+			<div className="left flex">
```
Comment (xhebox) [inline packages/website/app/components/headerNav.tsx:16]: Follow HTML5 spec, use `<li>` inside `<ul>`. No `div > li`, no `div > div > li`.

File: packages/website/app/types/articleBox.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+export type IArticle = {
+	id:number,
+	type:string,
+	name:string|null,
+	title:string,
+	content:string,
+	data:string,
+	createTime:Date,
+}|null
+export type ILoaderData = {
```
Comment (xhebox) [inline packages/website/app/types/articleBox.ts:19]: Not used `ILoaderData`

File: packages/website/app/routes/home.tsx [hunk 6]
```diff
@@ -1,36 +1,101 @@
...
+
+import { ArticleBox } from '~/components/articleBox/articleBox'
+import { db } from '~/utils/db.server'
+
+export const meta: MetaFunction = () => ({
+	title: 'Home',
+})
+
+export const handle = {
+	breadcrumb: () => <Link to="/home">Home</Link>,
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:19]: Refactor your code, you don't `() => <Link>`, you just need a title. `/home` can be computed from `match`.

I recommended you export a field called `attributes` since [mdx] will have such a field:

```
---
title: My First Post
---

# Example Markdown Post

You can reference your frontmatter data through "attributes". The title of this post is {attributes.title}!
```

File: packages/website/app/routes/home.tsx [hunk 7]
```diff
@@ -1,36 +1,101 @@
...
+}
+export const loader: LoaderFunction = async ({ request }) => {
+	const index =
+    parseInt(new URL(request.url).searchParams.get('index') as string) || 0
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:23]: Why do we have `index`? Do you want to support paging on an infinite scroll page? If so, I recommended https://remix.run/docs/en/v1/guides/data-loading#route-params. `route params` are easier than `query parameters`.

File: packages/website/app/routes/home.tsx [hunk 8]
```diff
@@ -1,36 +1,101 @@
...
+			createTime: true,
+		},
+	})
+	return data
+}
+
+
+export default function Home() {
+	const fetcher = useFetcher()
+	const [index, setIndex] = useState(0)
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:47]: What if `home?index=2`, you start the infinite scroll from 1? The index needs to be passed from loader to here.

File: packages/website/app/routes/home.tsx [hunk 9]
```diff
@@ -1,36 +1,101 @@
...
+	const [articleBoxList, setArticleBoxList] = useState<IArticleBox[]>([])
+	const [newData, setNewData] = useState({})
+	const [end,setEnd] = useState(false)
+	useEffect(() => {
+		fetcher.load(`/home?index=${index}`)
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:52]: Get some help, you have `useLoaderData`, https://remix.run/docs/en/v1/guides/data-loading#basics

Or even better, typed `useLoaderData` https://github.com/sergiodxa/remix-utils#useloaderdata

File: packages/website/app/routes/home.tsx [hunk 10]
```diff
@@ -1,36 +1,101 @@
...
+	}, [index])
+	useEffect(() => {
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:54]: Reminder, you can use `useEffect(async () => {})` if you don't care about the `dispose` of `useEffect`. Also, `fetch.on` is available. There is no need to write multiple `useEffect`, it only makes the page slower.

File: packages/website/app/routes/home.tsx [hunk 11]
```diff
@@ -1,36 +1,101 @@
...
+		const newArticleBoxList = articleBoxList
+		newArticleBoxList.push(fetcher.data)
+		setArticleBoxList(newArticleBoxList)
+	}, [fetcher.data])
+	useEffect(() => {
+		intersectionObserverFunc()
+	}, [newData])
+	const intersectionObserverFunc = () => {
+		let newIndex = index
+		const articleBoxListSentry = document.querySelector('.article-box-list-sentry') as Element
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:70]: Check `https://reactjs.org/docs/refs-and-the-dom.html#creating-refs`, document is client only code. It should be avoided if possible.

File: packages/website/app/routes/home.tsx [hunk 12]
```diff
@@ -1,36 +1,101 @@
...
+
+	return (
+		<div className="home-container w-screen  min-h-screen">
+			<div className="top h-3xl w-100vw mb-20 home-top-img ">
+			</div>
+			<div className="mx-auto mb-300px" w-w="1024px">
+				<h1 w-text="60px">探索IOClub首页</h1>
+				<p>MemberList</p>
+				<div>
+					{articleBoxList &&
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:89]: Ensure `articleBoxList` existed by a empty `[]` value.

File: packages/website/app/routes/home.tsx [hunk 13]
```diff
@@ -1,36 +1,101 @@
...
+            articleBoxList.map(articleBox=> articleBox&&<ArticleBox key={articleBox.id} data={articleBox}/>)}
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:90]: Same, ensure `articleBox` is not nil.

File: packages/website/app/routes/home.tsx [hunk 14]
```diff
@@ -1,36 +1,101 @@
...
+		let newIndex = index
+		const articleBoxListSentry = document.querySelector('.article-box-list-sentry') as Element
+		const intersectionObserver = new IntersectionObserver(function (entries) {
+			if (entries[0].intersectionRatio > 0) {
+				newIndex += 1
+				setIndex(newIndex)
+				intersectionObserver.unobserve(articleBoxListSentry)
+			}
+		})
+		intersectionObserver.observe(articleBoxListSentry)
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:78]: Registering is expensive, reuse the observer. And remove the oberver on page unload, check https://stackoverflow.com/questions/71242012/how-to-cleanup-useref-in-useeffect for some examples of `useEffect` and `dispose`.

File: packages/website/app/routes/home.tsx [hunk 15]
```diff
@@ -1,36 +1,101 @@
...
+	const fetcher = useFetcher()
+	const [index, setIndex] = useState(0)
+	const [articleBoxList, setArticleBoxList] = useState<IArticleBox[]>([])
+	const [newData, setNewData] = useState({})
+	const [end,setEnd] = useState(false)
+	useEffect(() => {
+		fetcher.load(`/home?index=${index}`)
+	}, [index])
+	useEffect(() => {
+		if (fetcher.data == undefined) {
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:55]: Judging by `articles.length == 0` is enough. No need for `end`.

File: packages/website/app/routes/home.tsx [hunk 16]
```diff
@@ -1,36 +1,101 @@
...
+					{articleBoxList &&
+            articleBoxList.map(articleBox=> articleBox&&<ArticleBox key={articleBox.id} data={articleBox}/>)}
+				</div>
+			</div>
+			<div
+				className="bottom article-box-list-sentry"
+				w-h="300px"
+				w-text="black center"
+				w-align="middle"
+			>{end?'END':''}</div>
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:98]: use `{end ? <div> : ''}`, that is wrapping `<div>` inside it. We can render one less element in DOM.

Comment (lyra-planet) [reply packages/website/app/routes/home.tsx:98]: fiexd

File: packages/website/app/root.tsx [hunk 17]
```diff
@@ -43,11 +49,17 @@ export default function App() {
...
+				w-justify="center"
+				w-p="0"
+				w-m="0"
+				w-text="#747474"
+			>
 				<HeaderNav />
-				<main w-flex="grow">
-					<Outlet />
-				</main>
+				<Outlet />
```
[the remark this answers is not in the document]
Comment (lyra-planet) [reply packages/website/app/root.tsx:62]: fixed

File: packages/website/app/root.tsx [hunk 18]
```diff
@@ -43,11 +49,17 @@ export default function App() {
 				<Meta />
 				<Links />
 			</head>
-			<body w-flex="~ col" w-justify='center' w-p="0" w-m="0" w-min-h="screen" w-color="gray-800" w-bg="light-50">
+			<body
+				className="min-h-screen leading-normal font-thin scrollbar-"
```
[the remark this answers is not in the document]
Comment (lyra-planet) [reply packages/website/app/root.tsx:53]: Why cant leading-normal font-thin be the global default preference :(

File: packages/website/.env (hunk 1 above)
Comment (lyra-planet) [reply packages/website/.env:1]: I create the file ,but it cant work :<

  25    return json(data)
  26 }
  27 const getData = async (articleBoxId = 1) => {
→ 28    const data: IArticleBox = await db.articleBox.findUnique(
error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:10
   |
 9 |   provider = "sqlite"
10 |   url      = env("DATABASE_URL")
   |

Comment (xhebox) [reply packages/website/.env:1]: *DO NOT RESOLVE CONVERSATION* by yourself, if it is not solved/fixed definitely.

Check if `https://www.npmjs.com/package/dotenv` helps.

Comment (lyra-planet) [reply packages/website/.env:1]: like this DATABASE_URL="file:../../../../secret/iolab-env.db" ? :<

File: packages/website/app/routes/home.tsx (hunk 7 above)
Comment (lyra-planet) [reply packages/website/app/routes/home.tsx:23]: We need infinite scrolling !!! :)

Comment (lyra-planet) [reply packages/website/app/routes/home.tsx:23]: if I use params , infinite scrolling will broken :<

Comment (lyra-planet) [reply packages/website/app/routes/home.tsx:23]: or we can use /home?index=${index}/pageId

Comment (xhebox) [reply packages/website/app/routes/home.tsx:23]: Abstract another `resource route` under `/app/routes/api/articles`.

Comment (xhebox) [reply packages/website/app/routes/home.tsx:23]: Your api design sucks, plz refer graphql pagination model for some general design.

**Comment by lyra-planet** [comment]
Lyra is heartbroken&nbsp;







------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年10月11日(星期二) 晚上10:17
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; "State ***@***.***&gt;; 
主题: Re: [io-club/website] add unlimited scrolling list &amp;&amp; article pages &amp;&amp; script init-db &amp;&amp; removed most of the css (PR #28)





 
@xhebox commented on this pull request.
 
 
In packages/website/app/routes/home.tsx:
 &gt; +import { json } from ***@***.***/server-runtime' +import {useEffect, useState } from 'react' + +import { ArticleBox } from '~/components/articleBox/articleBox' +import { db } from '~/utils/db.server' + +export const meta: MetaFunction = () =&gt; ({ +	title: 'Home', +}) + +export const handle = { +	breadcrumb: () =&gt; <Link to="/home"&gt;Home</Link&gt;, +} +export const loader: LoaderFunction = async ({ request }) =&gt; { +	const index = +    parseInt(new URL(request.url).searchParams.get('index') as string) || 0  
Your api design sucks, plz refer graphql pagination model for some general design.

File: packages/website/app/styles/root.css [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+    border-radius: 4px;
+    text-decoration: none;
+    height: 30px;
+    padding: 3px 5px;
+    transition: all 0.3s;
+}
+
+a:hover {
+    text-decoration: underline;
+}
```
Comment (xhebox) [inline packages/website/app/styles/root.css:21]: Same, unocss is possible. And they are mostly local to `headerNav` and `articleBox`. Moreover `text-decoration: none` is removed by default.

File: packages/website/app/routes/math.mdx [hunk 20]
```diff
@@ -15,4 +15,5 @@ export const links = () => [
 
 # Hello MDX!
 
+# 2312312
```
Comment (xhebox) [inline packages/website/app/routes/math.mdx:18]: Unnecessary change

File: packages/website/app/routes/home.tsx [hunk 21]
```diff
@@ -1,36 +1,76 @@
...
+	)
+}
+const Component = () => {
+	const isHydrated = useHydrated()
+
+	if (isHydrated) {
+		return <Home />
+	}
+}
+export default Component
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:76]: What is wrong with this page that it must be rendered on client completely?

File: packages/website/app/components/headerNav.tsx (hunk 4 above)
Comment (xhebox) [reply packages/website/app/components/headerNav.tsx:15]: This is not resolved

File: packages/website/app/routes/home.tsx (hunk 6 above)
Comment (xhebox) [reply packages/website/app/routes/home.tsx]: Not resolved

File: packages/website/app/routes/article/$articleId.tsx [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+			createTime: true,
+		},
+	}) as IArticle
+	return articleData
+}
+export default function ArticlePage() {
+	const {article:{frontmatter,code},articleBox} = useLoaderData<LoaderData>()
+	const Component = useMemo(() => getMDXComponent(code), [code])
+	useEffect(()=>{
+		console.log(frontmatter)
```
Comment (xhebox) [inline packages/website/app/routes/article/$articleId.tsx:61]: Removed debugging message

File: packages/website/app/routes/math.mdx (hunk 20 above)
Comment (xhebox) [reply packages/website/app/routes/math.mdx:18]: Not resolved

File: packages/website/app/routes/article.tsx [hunk 23]
```diff
@@ -0,0 +1,16 @@
+import { Outlet } from '@remix-run/react'
+
+export const handle = {
+	breadcrumb: () => 'Article'
+}
```
[the remark this answers is not in the document]
Comment (xhebox) [reply packages/website/app/routes/article.tsx]: Not resolved

File: packages/website/app/components/articleBox/articleBox.tsx [hunk 24]
```diff
@@ -1,56 +1,72 @@
...
-			</div>
-		</div>
-	)
-}
+import type { IArticleBox } from '../../types/articleBox'
+
+import { Link } from '@remix-run/react'
+import { useEffect,useRef } from 'react'
+
+import AvatarUrl from '../../temporary/avatar.jpg'
```
[the remark this answers is not in the document]
Comment (xhebox) [reply packages/website/app/components/articleBox/articleBox.tsx]: Not resolved

**Comment by lyra-planet** [comment]
&nbsp;I will resolve it in next PR




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年10月16日(星期天) 中午12:29
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; "State ***@***.***&gt;; 
主题: Re: [io-club/website] add unlimited scrolling list &amp;&amp; article pages &amp;&amp; script init-db &amp;&amp; removed most of the css (PR #28)





 
@xhebox commented on this pull request.
 
 
In packages/website/app/routes/article.tsx:
 &gt; +export const handle = { +	breadcrumb: () =&gt; 'Article' +}  
Not resolved

File: packages/website/.env [hunk 25]
```diff
@@ -1 +1,2 @@
-PORT=3000
+DATABASE_URL="file:./dev.db"
```
Comment (xhebox) [inline packages/website/.env:1]: `DATABASE_URL` not needed, included in `.env.development`

File: packages/website/styles/root.css [hunk 26]
```diff
@@ -1,3 +1,5 @@
-@import '@unocss/reset/tailwind.css';
-@import '@kidonng/daisyui/index.css';
-@import 'uno.css';
+@import '@unocss/reset/tailwind.css';
+@import '@kidonng/daisyui/index.css';
+@import 'uno.css';
+@import 'global.css';
+@import 'register.css';
```
Comment (xhebox) [inline packages/website/styles/root.css:5]: Why `register.css`? Why not merge your global css into `root.css`

File: packages/website/unocss.config.ts [hunk 27]
```diff
@@ -1,26 +1,37 @@
...
+			display: 'block',
+			width: '0%;',
+			height: '4px;',
+			background: 'gray;',
+			transition: 'all 0.3s;'}],
+		['keyframes-slide-in', {}]
+	],
+	exclude: [
+		/uno.css/,
+	],
```
Comment (xhebox) [inline packages/website/unocss.config.ts:20]: Useless

File: packages/website/unocss.config.ts [hunk 28]
```diff
@@ -1,26 +1,37 @@
...
 	transformers: [
 		transformerDirectives(),
 	],
 	presets: [
+		presetDaisy(),
```
Comment (xhebox) [inline packages/website/unocss.config.ts:25]: Move it down. This does not matter.

File: patches/prisma@4.4.0.patch [hunk 29]
```diff
@@ -1,42 +1,42 @@
-diff --git a/build/index.js b/build/index.js
-index d10dbcd9ffe376112fe3a0e5aa68f12618a7c33d..bcbb9fa36b727911450202a3e23d3ee7a86e4fb7 100755
---- a/build/index.js
```
Comment (xhebox) [inline patches/prisma@4.4.0.patch:3]: Do not modify this file.

File: packages/website/app/styles/root.css [hunk 30]
```diff
@@ -0,0 +1,6236 @@
+/*
+1. Prevent padding and border from affecting element width. (https://github.com/mozdevs/cssremedy/issues/4)
```
Comment (xhebox) [inline packages/website/app/styles/root.css:2]: Seriously?

File: docker/Dockerfile [hunk 31]
```diff
@@ -1,7 +1,7 @@
-FROM node:16-alpine
-
-ENV PORT=3000
-ADD . /srv
-RUN cd /srv && npm -g install pnpm && pnpm install && pnpm run build && rm -rf node_modules packages/*/node_modules && pnpm install -P && pnpm store prune
-WORKDIR /srv/packsages/website
-ENTRYPOINT ["./node_modules/.bin/remix-serve", "run", "serve"]
+FROM node:16-alpine
```
Comment (xhebox) [inline docker/Dockerfile:1]: Set your CRLF changes. No changes in this file.

File: packages/website/prisma/seed.ts [hunk 32]
```diff
@@ -1,21 +1,94 @@
...
+	// 	where: {
+	// 		nick: 'Lyra',
+	// 	},
+	// })
+	await db.articleBox.deleteMany({
+		where: {
+			name: 'Lyra',
 		},
 	})
+	// await db.user.create({
```
Comment (xhebox) [inline packages/website/prisma/seed.ts:17]: How about removing useless comments?

File: packages/website/package.json [hunk 33]
```diff
@@ -17,30 +18,37 @@
...
 		"@unocss/preset-mini": "^0.45.29",
 		"@unocss/reset": "^0.45.29",
 		"cookie-signature": "^1.2.0",
 		"dotenv-cli": "^6.0.0",
+		"esbuild-register": "^3.3.3",
 		"fastify": "^4.8.1",
 		"fastify-plugin": "^4.2.1",
 		"fastify-racing": "^1.0.3",
-		"graphql": "^16.0.0",
+		"faunadb": "^4.7.0",
```
Comment (xhebox) [inline packages/website/package.json:42]: faunadb is removed.

File: packages/website/package.json [hunk 34]
```diff
@@ -17,30 +18,37 @@
...
+		"graphql": "^16.6.0",
+		"highlight.js": "^11.6.0",
 		"katex": "^0.16.2",
-		"mercurius": "^11.0.0",
+		"mercurius": "^11.2.0",
 		"nanoid": "^4.0.0",
 		"nexus": "^1.3.0",
 		"react": "^18.2.0",
 		"react-dom": "^18.2.0",
+		"react-katex": "^3.0.1",
```
Comment (xhebox) [inline packages/website/package.json:51]: Sample `math.mdx` and `react-katex` are removed in the master.

File: packages/website/package.json [hunk 35]
```diff
@@ -53,31 +61,34 @@
 		"@julr/unocss-preset-heropatterns": "^0.0.2",
 		"@kidonng/daisyui": "^2.31.0",
 		"@remix-run/dev": "^1.7.2",
+		"@remix-run/node": "^1.7.2",
```
Comment (xhebox) [inline packages/website/package.json:64]: `@remix-run/node` is not used anywhere.

File: packages/website/package.json [hunk 36]
```diff
@@ -53,31 +61,34 @@
...
 		"nodemon": "^2.0.20",
 		"postcss": "^8.4.18",
 		"postcss-cli": "^10.0.0",
 		"postcss-import": "^15.0.0",
 		"postcss-nesting": "^10.2.0",
 		"postcss-pseudo-class-enter": "^5.0.0",
 		"postcss-url": "^10.1.3",
-		"prisma": "^4.5.0",
+		"prisma": "^4.4.0",
+		"prop-types": "^15.8.1",
```
Comment (xhebox) [inline packages/website/package.json:78]: What is `prop-types` for?

File: packages/website/app/routes/login.tsx [hunk 37]
```diff
@@ -18,15 +18,15 @@ export default function Screen() {
...
+export const action: ActionFunction = async ({ request, context }:any) => {
 	return await context.auth.authenticate('user-pass', request, {
 		successRedirect: '/',
 		failureRedirect: '/register',
 	})
 }
 
 
-export const loader: LoaderFunction = async ({ request, context }) => {
+export const loader: LoaderFunction = async ({ request, context }:any) => {
```
Comment (xhebox) [inline packages/website/app/routes/login.tsx:29]: Makes type inference unworking, not acceptable.

File: packages/website/app/routes/index.tsx [hunk 38]
```diff
@@ -5,33 +5,36 @@ import { useEffect, useRef } from 'react'
...
 	const reactCanvas = useRef(null)
 	const navigate = useNavigate()
 
 	useEffect(() => {
 		let cancelled = false
 
 		const render = async() => {
+			//@ts-ignore
 			const { Analyser, ArcRotateCamera, AssetsManager, Camera, Color4, CreateGround, DynamicTexture, Effect, Engine, HemisphericLight, MeshBuilder, RawTexture, Scene, ShaderMaterial, Sound, SoundTrack, StandardMaterial, Texture, Vector2, Vector3, Vector4 } = await import('@babylonjs/core')
+			//@ts-ignore
```
Comment (xhebox) [inline packages/website/app/routes/index.tsx:37]: Not reporting errors for me, not acceptable changes.

File: packages/website/app/routes/index.tsx [hunk 39]
```diff
@@ -5,33 +5,36 @@ import { useEffect, useRef } from 'react'
...
 
 interface DemoProps {
-	antialias?: EngineOptions['antialias']
-	adaptToDeviceRatio?: EngineOptions['adaptToDeviceRatio']
-	engineOptions?: EngineOptions
-	sceneOptions?: SceneOptions
+	antialias?: EngineOptions['antialias'];
+	adaptToDeviceRatio?: EngineOptions['adaptToDeviceRatio'];
+	engineOptions?: EngineOptions;
+	sceneOptions?: SceneOptions;
```
Comment (xhebox) [inline packages/website/app/routes/index.tsx:18]: Semicolon unnecessary, if your linter reports errors, tell me. I'll try to modify eslint settings.

File: packages/website/app/routes/home.tsx [hunk 40]
```diff
@@ -1,37 +1,79 @@
...
+
+const Component = () => {
+	const isHydrated = useHydrated()
+
+	if (isHydrated) {
+		return <Home />
+	}
+}
+
+export default Component
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:79]: Why it is changed back?

File: packages/website/app/routes/home.tsx [hunk 41]
```diff
@@ -1,37 +1,79 @@
...
 			</div>
-			<div className="bottom">
+			<div
+				className="article-box-list-sentry"
+				ref={scroll}
+				w-h="300px"
+				w-text="black center"
+				w-align="middle"
+			>
+				{more ? '' : <div>END</div>}
```
Comment (xhebox) [inline packages/website/app/routes/home.tsx:65]: Same, more outside, div inside to eliminate the DOM element

File: packages/website/app/routes/article/$articleId.tsx [hunk 42]
```diff
@@ -0,0 +1,43 @@
...
+		},
+		select: {
+			id: true,
+			type: true,
+			data: true,
+			name: true,
+			title: true,
+			content: true,
+			createTime: true,
+		},
```
Comment (xhebox) [inline packages/website/app/routes/article/$articleId.tsx:30]: If you need al data, just don't pass `select` prop.

File: packages/website/app/routes/article/$articleId.tsx [hunk 43]
```diff
@@ -0,0 +1,43 @@
...
+	}) as IArticle
```
Comment (xhebox) [inline packages/website/app/routes/article/$articleId.tsx:31]: Unncessary type cast

File: packages/website/app/root.tsx [hunk 44]
```diff
@@ -24,6 +25,10 @@ export const links: LinksFunction = () => {
 			rel: 'stylesheet',
 			href: unoCssUrl,
 		},
+		{
+			rel: 'stylesheet',
+			href: globalCssUrl,
```
Comment (xhebox) [inline packages/website/app/root.tsx:30]: I still insist this should be component-wise css. It should be included globally.

File: packages/website/app/components/headerNav.tsx [hunk 45]
```diff
@@ -1,29 +1,54 @@
...
+			w-top="0"
+			w-items="center"
+		>
+			<div className="flex">
+				{matches
+					.filter((match) => match.handle && match.handle.breadcrumb)
+					.map((match, index) => {
+						return (
+							<ul className="flex" key={match.id}>
+								<li>{match.handle?.breadcrumb(match)}{match.data ? ('/'+match.data.article.frontmatter.title):''}</li>
```
Comment (xhebox) [inline packages/website/app/components/headerNav.tsx:23]: I want to see only one `{match.handle?.breadcrumb(match)}` or `{match.data ? ('/'+match.data.article.frontmatter.title):''}`.

That is, a consistent method to get the title.
